src/datasources/erc721.ts

```diff
@@ -99,3 +106,29 @@ export function handleApprovalForAll(event: ApprovalForAllEvent): void {
 		// 	ev.save()
 	}
 }
+
+export function handleMetadataUpdate(event: MetadataUpdateEvent) : void {
+	let contract = fetchERC721(event.address)
+	if (contract != null) {
+		let erc721       = IERC721.bind(Address.fromBytes(contract.id))
+
+		let token        = fetchERC721Token(contract, event.params._tokenId)
+		let try_tokenURI = erc721.try_tokenURI(event.params._tokenId)
+		token.uri        = try_tokenURI.reverted ? '' : try_tokenURI.value
```

Comment:
Feels weird defaulting to an empty string. This shouldn't revert afaik.

Reply:
It shouldn't, but it can ... and if it does, then the subgraph fails.

In the case of a subgraph that indexes all ERC-721 contracts on a chain, it easy for an attacker to build a malicious contract that triggers that event, and then have the `tokenURI(uint256)` function revert. In that case, we want the subgraph to not fail completelly.